column_index.go

```diff
@@ -87,46 +87,55 @@ func (f *formatColumnIndex) IsDescending() bool {
 type fileColumnIndex struct{ chunk *fileColumnChunk }
 
 func (i fileColumnIndex) NumPages() int {
-	return len(i.chunk.columnIndex.NullPages)
+	return len(i.columnIndex().NullPages)
 }
 
 func (i fileColumnIndex) NullCount(j int) int64 {
-	if len(i.chunk.columnIndex.NullCounts) > 0 {
-		return i.chunk.columnIndex.NullCounts[j]
+	index := i.columnIndex()
+	if len(index.NullCounts) > 0 {
+		return index.NullCounts[j]
 	}
 	return 0
 }
 
 func (i fileColumnIndex) NullPage(j int) bool {
-	return len(i.chunk.columnIndex.NullPages) > 0 && i.chunk.columnIndex.NullPages[j]
+	return i.nullPage(j, i.columnIndex())
+}
+
+func (i fileColumnIndex) nullPage(j int, index *format.ColumnIndex) bool {
+	return len(index.NullPages) > 0 && index.NullPages[j]
```

Comment:
How about making this a function since we never access the receiver?

```go
func nullPage(index *format.ColumnIndex, i int) bool {
    return len(index.NullPages) > 0 && index.NullPages[i]
}
```

Reply:
Done. There was a naming conflict since `nullPage` is already a top-level type. So I named the function `isNullPage`.